### PR TITLE
Ga powered popular articles

### PR DESCRIPTION
**Pull request by baphled**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -43,6 +43,10 @@ gem 'rails-observers', '~> 0.1.2'
 gem 'ckeditor', github: 'galetahub/ckeditor'
 gem 'autoprefixer-rails'
 
+gem 'oauth2', '1.0.0'
+gem 'google-api-client', '0.7.1'
+gem 'legato', '0.4.0'
```
Comment (esbel) [inline Gemfile]: Do you need these gems to be exactly in specified versions?

In other words, would it break if I updated google-api-client to a more recent version?

Comment (baphled) [inline Gemfile]: CMS uses similar functionality, I just pulled those gems from there.

File: app/views/shared/_footer.html.erb [hunk 2]
```diff
@@ -2,28 +2,34 @@
...
+            <time class="l-footer-link__list-item-time">24th January</time>
+          </li>
+          <li class="l-footer-link__list-item">
+            <a href="/the-online-tax-return-deadline-is-looming-missing-it-could-be-costly">The online tax return deadline is looming. Missing it could be costly…</a>
+            <time class="l-footer-link__list-item-time">23rd January</time>
+          </li>
+          <li class="l-footer-link__list-item">
+            <a href="/buying-a-house-why-you-shouldnt-change-your-mobile">Buying a house? Why you shouldn’t change your mobile</a>
+            <time class="l-footer-link__list-item-time">9th January</time>
+          </li>
```
Comment (tomas-stefano) [inline app/views/shared/_footer.html.erb]: These are fixed articles?

File: spec/lib/popular_article_spec.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  let!(:blog) { create(:blog) }
+
+  subject { described_class }
+
+  describe '#find' do
+    before(:each) do
+      allow(GoogleAnalytics::API).to receive(:fetch_article_page_views).and_return(popular_articles)
+    end
+
+    let(:popular_articles) {
```
Comment (esbel) [inline spec/lib/popular_article_spec.rb]: use do/end for multiline blocks https://github.com/bbatsov/ruby-style-guide#single-line-blocks

File: app/views/shared/_footer.html.erb (hunk 2 above)
Comment (baphled) [inline app/views/shared/_footer.html.erb]: Nice catch, not sure why git overwrote that.

File: spec/lib/popular_article_spec.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
+require File.join(Rails.root, 'lib', 'popular_article')
+
+describe PopularArticle do
+  let!(:blog) { create(:blog) }
+
+  subject { described_class }
+
+  describe '#find' do
```
Comment (tomas-stefano) [inline spec/lib/popular_article_spec.rb]: I think you mean '.find' ;).

File: spec/lib/popular_article_spec.rb (hunk 3 above)
Comment (baphled) [inline spec/lib/popular_article_spec.rb]: I'm not sure about that with before blocks.

In line with making small changes there's normally higher churn with spec before blocks.

File: spec/lib/popular_article_spec.rb (hunk 4 above)
Comment (baphled) [inline spec/lib/popular_article_spec.rb]: nah, `#` is consistent with the rest of the specs :)

Comment (tomas-stefano) [inline spec/lib/popular_article_spec.rb]: Ow, I see. But even with that, I would use the Ruby documentation convention of . (or ::) when referring to a class method's name and # when referring to an instance method's name. And I would move the subject line inside of the describe block too. Wdyt?

Comment (baphled) [inline spec/lib/popular_article_spec.rb]: In an ideal world I'd do so though the code base needs some love in generally.

Maybe that's the best time to look at such things :).

**Comment by tomas-stefano** [comment]
Besides of the stylistic comment, everything looks good now :+1:

**Comment by rnalexander** [comment]
:+1:

**Pull request merged**